Description: Handle the G_HOME environment variable to override the passwd entry
 This will allow to fix various kinds of build failures due to restricted
 build environments.
Author: Josselin Mouette <joss@debian.org>
Origin: vendor

Index: glib_2.33.14-SprezzOS1/glib/gutils.c
===================================================================
--- glib_2.33.14-SprezzOS1.orig/glib/gutils.c	2012-09-17 17:07:08.000000000 -0400
+++ glib_2.33.14-SprezzOS1/glib/gutils.c	2012-10-03 21:46:56.672109474 -0400
@@ -700,11 +700,14 @@
     }
 #endif	/* !G_OS_WIN32 */
   
+  g_home_dir = g_strdup (g_getenv ("G_HOME"));
+  
 #ifdef G_OS_WIN32
   /* We check $HOME first for Win32, though it is a last resort for Unix
    * where we prefer the results of getpwuid().
    */
-  g_home_dir = g_strdup (g_getenv ("HOME"));
+  if (!g_home_dir)
+    g_home_dir = g_strdup (g_getenv ("HOME"));
 
   /* Only believe HOME if it is an absolute path and exists */
   if (g_home_dir)
@@ -1000,6 +1003,11 @@
  *      homedir = g_get_home_dir (<!-- -->);
  * ]|
  *
+ * However, to allow changing this value for testing and development
+ * purposes, the value of the <envar>G_HOME</envar> environment 
+ * variable, if set, will override the <filename>passwd</filename>
+ * entry.
+ *
  * Returns: the current user's home directory
  */
 const gchar *
